openshift_scalability/ci/scripts/pbench-controller-start-node-vertical.sh

```diff
@@ -0,0 +1,296 @@
+scp root@${MASTER_PUBLIC_DNS}:/opt/pbench-agent/config/pbench-agent.cfg /opt/pbench-agent/config/pbench-agent.cfg
+echo -e "Checking the newly copied /opt/pbench-agent/config/pbench-agent.cfg file on this host: \n"
+cat /opt/pbench-agent/config/pbench-agent.cfg
+
+
+echo -e "\nRunning pbench-stop-tools and pbench-clear-results on this Test Client host: \n"
+pbench-stop-tools
+pbench-clear-results
+
+
+# get the internal node ip addresses now that we have the /root/.kube/config copied over:
+## NODES=$(oc get nodes | grep -v "SchedulingDisabled" | grep -v NAME | awk '{print $1}'  | xargs)
+## NODES are used when we want to register pbench from scratch
+NODES=$(oc get nodes | grep -v "master" | grep -v NAME | awk '{print $1}'  | xargs)
+
+ALL_NODES_INTERNAL=$(oc get nodes | grep -v NAME | awk '{print $1}'  | xargs)
+## MASTER_INTERNAL_IP=$(oc get nodes | grep "SchedulingDisabled" | awk '{print $1}')
+MASTER_INTERNAL_IP=$(oc get nodes --no-headers -l node-role.kubernetes.io/master=true | cut -f1 -d" ")
+
+echo -e "\nChecking node config master node:"
+ssh ${MASTER_PUBLIC_DNS} "cat /etc/origin/node/node-config.yaml"
+
+echo -e "\nNodes other than masters: ${NODES}, and etcds: ${STANDALONE_ETCDS_PRIVATE_DNS}"
+echo -e "All Nodes internal ip addresses: ${ALL_NODES_INTERNAL}"
+echo -e "Master internal ip address: ${MASTER_INTERNAL_IP}"
+
+# checking if the pbench directories were created
+echo -e "\nChecking if the pbench directories were previously created by Flexy"
+
+for i in ${ALL_NODES_INTERNAL} ${STANDALONE_ETCDS_PRIVATE_DNS} ; do
+  echo -e "\nssh to $i: "
+  ssh $i "ls -ltr /var/lib/pbench-agent/ ; ls -ltr /var/lib/pbench-agent/tools-default"
+done
+
+
+
+echo -e "\nClearing past pbench results on all nodes:"
+
+# Run on Test Client:
+for node in ${ALL_NODES_INTERNAL} ${STANDALONE_ETCDS_PRIVATE_DNS} ; do
+  echo -e "\nssh to $node: "
+  ssh ${node} "pbench-clear-results"
+done
+
+sleep 10
+
+####################################################################
+# Using Flexy now with pbench_controller_count set to 1 to creates a jump node and
+# runs pbench-ansible to register all nodes, so we can skip the pbench-register steps
+# unless we want to over-ride this and re-register pbench
+# PBENCH_REGISTER=false; # by default
+
+if [ "${PBENCH_REGISTER}" == "true" ]; then
+
+  echo -e "Running pbench-kill-tools pbench-clear-tools on this Test Client host: \n"
+  pbench-kill-tools
+  pbench-clear-tools
+
+  echo -e "\nClearing past pbench tools setup and pbench results on all nodes:"
+
+  # Run on Test Client:
+  for node in ${ALL_NODES_INTERNAL} ${STANDALONE_ETCDS_PRIVATE_DNS} ; do
+    echo -e "\nssh to $node: "
+    ssh ${node} "pbench-clear-tools; pbench-clear-results"
+  done
+
+  sleep 10
+
+  echo -e "\nSSH to test Client and register pbench on all nodes:"
+
+
+  # pbench register master node first.  Handle case where we have more than one Master later, add for loop.
```

Comment:
i thought the registration was done by the pbench ansible job?

Reply:
Default is using pbench-ansible for registration.  This is an optional step that allows you to re-register pbench if you want to just run it from master when you don't have a pbench-controller node or test client available.